### ISSUE TITLE
confusing style for highway=pedestrian

### ISSUE THREAD

**Issue by mnalis** [ENT8R/streetcomplete-mapstyle#119]
`highway=pedestrian` has strange style, with divider line in the middle, which makes it look like two streets.

![small_Screenshot_20211223_165022_de westnordost streetcomplete mn debug](https://user-images.githubusercontent.com/156656/147264257-396a7224-32af-4f1f-910e-f6190969bea4.jpg)

It is confusing to users, eg. https://www.openstreetmap.org/note/2259724 or https://www.openstreetmap.org/note/2974300

Perhaps it could be drawn to look more like regular footway?

**Comment by westnordost** [comment]
Sounds like a bug. I can not remember there is some explicit code in the style that would do this.

**Comment by mnalis** [comment]
Looking around at https://streetcomplete.github.io/streetcomplete-mapstyle/?provider=jawg&style=dark, the problematic _"line in the middle of pedestrian street"_ is visible in both `dark` and `light` mode, but **not** in `satellite` mode, where it looks normal.

**Comment by Helium314** [comment]
The responsible part is https://github.com/streetcomplete/streetcomplete-mapstyle/blob/2ecce80b365b2eea1e1351ad3abe5f8b355f9a5b/layers/land.yaml#L123-L126
which also draws the "double" outline of squares.
IMO things also look quite ok without this part

**Comment by westnordost** [comment]
Hm well, the outlining works correctly with buildings, so maybe there is a bug.

**Comment by mnalis** [comment]
hm, that `function()` is inside `pedestrian-areas:` yaml section, and [highway=pedestrian](https://wiki.openstreetmap.org/wiki/Tag:highway=pedestrian) (which this issue is about) can be used not only as a **way**, but _also_ as an **area**. Could that be related to the problem?

i.e. that style should be applied only to areas, and not to regular (non-closed) ways?

**Comment by Helium314** [comment]
> Hm well, the outlining works correctly with buildings, so maybe there is a bug.

It looks like outlines of squares are displayed as roads. When excluding pedestrian roads from normal road style, the outlines of squares are nice, but then of course non-area pedestrian roads are not displayed properly (only as line with `square_outline_color`).

**Comment by michaelblyons** [comment]
> hm, that `function()` is inside `pedestrian-areas:` yaml section, and [highway=pedestrian](https://wiki.openstreetmap.org/wiki/Tag:highway=pedestrian) (which this issue is about) can be used not only as a **way**, but _also_ as an **area**. Could that be related to the problem?
> 
> i.e. that style should be applied only to areas, and not to regular (non-closed) ways?

The extra bubbling outward of the area is also ugly in the places where I encounter it. Mostly around swimming pools, but a couple other places too.

**Comment by matkoniecz** [comment]
@Joxit Is there any possibility to distinguish "outline of highway=pedestrian area=yes area" and "highway=pedestrian line"?

From what I see Jawg data is not allowing this.

**Comment by matkoniecz** [comment]
I tried removing

```
                lines:
                    color: global.square_outline_color
                    width: function () { return 1/4 * Math.log($zoom); }
                    order: 30
```

and I think that with current limitations it is overall an improvement. 

Though one of results is no longer displaying minor inner borders of pedestrian areas, like https://www.openstreetmap.org/way/304011759

(transparency difference is an unrelated issue)
![screen11](https://user-images.githubusercontent.com/899988/193240353-6e68d131-740c-487e-981b-534fbab47ea9.png)

**Comment by westnordost** [comment]
@matkoniecz what data is exactly in the jawg tile set is documented here: 

https://www.jawg.io/docs/apidocs/maps/streets-v2/

Note the map on top, you are able to zoom in and inspect exactly what tags are set on the individual objects.

**Comment by matkoniecz** [comment]
> Note the map on top, you are able to zoom in and inspect exactly what tags are set on the individual objects.

I did this to confirm that there is nothing allowing distinguishing them.

![screen12](https://user-images.githubusercontent.com/899988/193254680-32a91618-3759-4522-8937-749ee07d0a96.png)
![screen14](https://user-images.githubusercontent.com/899988/193254797-f2b76813-3f54-4ff3-abc8-e5cb6e373d9f.png)

(`oneway` and presence of names is not reliable indicator)

**Comment by Joxit** [comment]
Hi there, so you want to distinguish a Line from a Polygon ? With MapLibre GL, the style spec allow us to apply style on LineString or Polygon, is that you are looking for ?

**Comment by mnalis** [comment]
@Joxit yes, the intention is to distinguish from `highway=pedestrian` way and `highway=pedestrian` polygon (i.e. closed way, area), which currently seems to be impossible.
